behat/tests/bootstrap/UserManagement/UserManagementTrait.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace DigidepsBehat\UserManagement;
+
+use Behat\Gherkin\Node\TableNode;
+use Behat\Mink\Session;
+use Symfony\Component\HttpFoundation\Response;
+use function GuzzleHttp\Psr7\build_query;
+
+trait UserManagementTrait
+{
+    /**
+     * it's assumed you are logged as an admin and you are on the admin homepage (with add user form).
+     *
+     * @Given the following clients exist and are attached to deputies:
+     */
+    public function clientsExist(TableNode $table)
+    {
+        foreach ($table as $inputs) {
+            $query = build_query($inputs);
+            $this->visitAdminPath("/admin/fixtures/createClientAttachDeputy?$query");
+        }
+    }
+
+    /**
+     * Requires a logged in admin user to call
+     *
+     * @Given I am viewing the edit user page for :userEmail
+     */
+    public function viewEditUserPageFor($userEmail)
+    {
+        $this->visitAdminPath("/admin/fixtures/getUserIDByEmail/" . strtolower($userEmail));
+
+        /** @var Session $session */
+        $session = $this->getSession();
+
+        if ($session->getStatusCode() !== Response::HTTP_OK) {
+            throw new \Exception($session->getPage()->getContent());
+        }
+
+        $id = $session->getPage()->getContent();
+        $this->visitAdminPath("/admin/edit-user?filter=$id");
```

Comment:
For future discussion: I'm not clear how far we want to take these fixture help functions if the functionality is achievable through the UI.

For example, this could also be achieved by opening `/admin`, searching for the email address and clicking `behat-link-user-{condensedEmail}`.

Reply:
True - it would be interesting to see if there's any difference in speed between API calls and page renders. I always tend to prefer going straight to the source if possible rather than driving a page but good to chat through pros and cons.